partiql-spi/src/main/java/org/partiql/spi/errors/Classification.java

```diff
@@ -0,0 +1,107 @@
+package org.partiql.spi.errors;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.spi.PEnum;
+
+/**
+ * TODO
+ */
+public final class Classification extends PEnum {
+
+    private Classification(int code) {
+        super(code);
+    }
+
+    /**
+     * TODO
+     */
+    public static final int UNKNOWN = 0;
+
+    /**
+     * TODO
+     * @return TODO
+     */
+    @NotNull
+    public static Classification UNKNOWN() {
+        return new Classification(UNKNOWN);
+    }
+
+    /**
+     * TODO
+     */
+    public static final int SYNTAX = 1;
+
+    /**
+     * TODO
+     * @return TODO
+     */
+    @NotNull
+    public static Classification SYNTAX() {
+        return new Classification(SYNTAX);
+    }
+
+    /**
+     * TODO
+     */
+    public static final int SEMANTIC = 2;
+
+    /**
+     * TODO
+     * @return TODO
+     */
+    @NotNull
+    public static Classification SEMANTIC() {
+        return new Classification(SEMANTIC);
+    }
+
+    /**
+     * TODO
+     */
+    public static final int COMPILATION = 3;
+
+    /**
+     * TODO
+     * @return TODO
+     */
+    @NotNull
+    public static Classification COMPILATION() {
+        return new Classification(COMPILATION);
+    }
+
+    /**
+     * TODO
+     */
+    public static final int EXECUTION = 4;
```

Comment:
What are some examples of compilation vs execution errors? 

In a recent message we were talking about how lazy datums hide execution problems. I'm not sure I understand how we might get an execution PError rather than something akin to a "runtime exception" when materializing a lazy datum.

Reply:
Perhaps some driver in partiql-lang would take a listener an it might emit execution errors? I somewhat think partiql-eval should have an evaluator driver that can emit them. By driver, I just mean something that materializes Datums. So a customer may choose to use the lazy datum, or just invoke an eval() which actually performs the work and accepts a listener. Does that make sense?

Reply:
> What are some examples of compilation vs execution errors?

Compilation errors might happen if the plan is referencing a custom physical operator that doesn't exist, or if a feature is not supported for execution (yet). It describes taking the process of taking the input plan and outputting an executable. There may be many types of errors at this stage.

Execution errors may be emitted during actual execution, such as data exceptions. I actually discuss this briefly in the usage guide, but the idea is that all components will standardize the fact the the only JVM exception that these components will see is the `PErrorListenerException`. Though, for exceptions that arise during the execution of the prepared statement, I detailed an `EvaluationException` that essentially wraps a `PError`.